packages/at_auth/lib/src/enroll/at_enrollment_impl.dart

```diff
@@ -0,0 +1,213 @@
+      throw AtEnrollmentException(
+          'Enrollment response from server: $enrollResult');
+    }
+    enrollResult = enrollResult.replaceFirst('data:', '');
+    var enrollJson = jsonDecode(enrollResult);
+    var enrollmentIdFromServer = enrollJson[AtConstants.enrollmentId];
+
+    // Close the connection after enrollment operation is completed.
+    await (atLookUp as AtLookupImpl).close();
+
+    return EnrollResponse(enrollmentIdFromServer,
+        getEnrollStatusFromString(enrollJson['status']));
+  }
+
+  @override
+  Future<AtEnrollmentResponse> manageEnrollmentApproval(
+      AtEnrollmentRequest atEnrollmentRequest, AtLookUp atLookUp) {
+    switch (atEnrollmentRequest.enrollOperationEnum) {
+      case EnrollOperationEnum.approve:
+        return _handleApproveOperation(atEnrollmentRequest, atLookUp);
+      case EnrollOperationEnum.deny:
+        return _handleDenyOperation(atEnrollmentRequest, atLookUp);
+      default:
+        throw AtEnrollmentException('Enrollment operation is not provided');
+    }
+  }
+
+  Future<AtEnrollmentResponse> _handleApproveOperation(
+      AtEnrollmentRequest atEnrollmentRequest, AtLookUp atLookUp) async {
+    // Decrypt the encrypted APKAM Symmetric key
+    var defaultEncryptionPrivateKey = RSAPrivateKey.fromString(atLookUp
+        .atChops!.atChopsKeys.atEncryptionKeyPair!.atPrivateKey.privateKey);
+    var apkamSymmetricKey = defaultEncryptionPrivateKey
+        .decrypt(atEnrollmentRequest.encryptedAPKAMSymmetricKey!);
+    atLookUp.atChops?.atChopsKeys.apkamSymmetricKey = AESKey(apkamSymmetricKey);
+
+    String command = 'enroll:approve:${jsonEncode({
+          'enrollmentId': atEnrollmentRequest.enrollmentId,
+          'encryptedDefaultEncryptedPrivateKey': atLookUp.atChops
+              ?.encryptString(
+                  atLookUp.atChops!.atChopsKeys.atEncryptionKeyPair!
+                      .atPrivateKey.privateKey,
+                  EncryptionKeyType.aes256,
+                  keyName: 'apkamSymmetricKey',
+                  iv: AtChopsUtil.generateIVLegacy())
+              .result,
+          'encryptedDefaultSelfEncryptionKey': atLookUp.atChops
+              ?.encryptString(
+                  atLookUp.atChops!.atChopsKeys.selfEncryptionKey!.key,
+                  EncryptionKeyType.aes256,
+                  keyName: 'apkamSymmetricKey',
+                  iv: AtChopsUtil.generateIVLegacy())
+              .result
+        })}';
+
+    String? enrollResponse =
+        await atLookUp.executeCommand('$command\n', auth: true);
+    enrollResponse = enrollResponse?.replaceAll('data:', '');
+    var enrollmentJsonMap = jsonDecode(enrollResponse!);
+    AtEnrollmentResponse enrollmentResponse = AtEnrollmentResponse(
+        enrollmentJsonMap['enrollmentId'],
+        _convertEnrollmentStatusToEnum(enrollmentJsonMap['status']));
+    // Close the connection after enrollment operation is completed.
+    await (atLookUp as AtLookupImpl).close();
```

Comment:
please revisit this logic..Since atLookup object is closed here, caller must not be using the atLookUp object after calling this method. Better to close atLookUp in the caller in a finally block

Reply:
Sure, will remove the call to close() method from this class.